tests/mir_test_framework/CMakeLists.txt

```diff
@@ -61,6 +61,12 @@ set_property(
     SOURCE udev_environment.cpp
     PROPERTY COMPILE_OPTIONS -Wno-variadic-macros)
 
+if ("libinput_environment.cpp:restrict" IN_LIST MIR_COMPILER_QUIRKS)
+  set_property(
+    SOURCE libinput_environment.cpp
+    PROPERTY COMPILE_OPTIONS -Wno-error=restrict)
+endif()
```

Comment:
This feels like too wide a net, should we take the compiler ID / version into account? This way we get feedback from other toolchains whether this becomes / stops being a problem?

Reply:
My idea here is to punt *when* to enable these quirks to configure time. I can modify `debian/rules` to additionally only enable these on gcc-12.

Reply:
> My idea here is to punt _when_ to enable these quirks to configure time. I can modify `debian/rules` to additionally only enable these on gcc-12.

Sure, that works, too.